### ISSUE TITLE
Sentinel Error values should not be overwritten

### ISSUE THREAD

**Issue by breml** [go-critic/go-critic#833]
Sentinel Error values (e.g. [`io.EOF`](https://golang.org/pkg/io/#pkg-variables)) are used as kind of constants (even though they are in fact public variables) and should therefore never be overwritten.

So go-critic should warn about assignments to public variables of type error if this does happen in a different package.

Consider the following code:

```
io.EOF = nil
```

With this, all checks for `if err == io.EOF` will result in pretty unexpected results.

**Comment by breml** [comment]
Results will be especially interesting, if someone overwrites things like [`rsa.ErrVerification`](https://golang.org/pkg/crypto/rsa/#pkg-variables).

**Comment by cristaloleg** [comment]
Maybe there are other non-error variables that shouldn't be changed. WDYT?

**Comment by quasilyte** [comment]
This check would be simpler with go-analysis framework.

We need to mark global `error` variables that have explicit initialization as sentinel errors and then warn on all assignments to them.

Giving a warning for any error-typed variable from another package might result in a false positive in case of some weird API that expects such global-reporting error passing or whatever. :)

**Comment by Antonboom** [comment]
Check is covered by https://golangci-lint.run/usage/linters/#reassign
